/*
 * hovergroup_proto
 *        File: geometry.proto
 *  Created on: Apr 14, 2014
 *      Author: Josh Leighton
 */

enum ShapeEnum {
	SQUARE = 1;
	TRIANGLE = 2;
	DIAMOND = 3;
	CIRCLE = 4;
	GATEWAY = 5;
	EFIELD = 6;
}

message VIEW_POINT {
	required double x = 1;
	required double y = 2;
	optional double z = 3;
	
	required string label = 4;
	optional string msg = 5;
	
	optional bool active = 6;
	optional string label_color = 8;
	optional string vertex_color = 9;
	optional int32 vertex_size = 10;
}

message VIEW_MARKER {
	required double x = 1;
	required double y = 2;
	
	required string label = 4;
	optional string msg = 5;
	
	optional ShapeEnum shape = 6;
	optional string primary_color = 7;
	optional string secondary_color = 8;
	optional int32 width = 9;
	optional bool active = 10;
}

message VIEW_SEGLIST {
	repeated double x = 1;
	repeated double y = 2;
	repeated double z = 3;
	
	required string label = 4;
	optional string msg = 5;
	
	optional int32 vertex_size = 6;
	optional int32 edge_size = 7;
	
	optional string edge_color = 8;
	optional string vertex_color = 9;
	optional bool active = 10;
}

message VIEW_POLYGON {
	required string label = 1;
	required string view_poly = 2;
}

message VIEW_RANGE_PULSE {

}

message VIEW_VECTOR {

}